src/test/CMakeLists.txt

```diff
@@ -159,14 +159,6 @@ if(ENABLE_WALLET)
 endif()
 
 if(WITH_MULTIPROCESS)
-  add_library(bitcoin_ipc_test STATIC EXCLUDE_FROM_ALL
-    ipc_test.cpp
-  )
-
-  target_capnp_sources(bitcoin_ipc_test ${PROJECT_SOURCE_DIR}
```

Comment:
Did you move these out of here because you'd otherwise get the following error:
```
*** Uncaught exception ***
capnp/compiler/node-translator.c++:1374: context: member.name = arg
kj/filesystem.c++:315: failed: expected parts.size() > 0 [0 > 0]; can't use ".." to break out of starting directory
stack: 59a492c93c94 59a492c93fc7 59a492c916b3 59a492a192f8 59a492b094db 59a492b08eb6 59a492b35fad 59a492b431c8 59a492b043ad 59a492b085a9 59a492b07e69 59a492b080d7 59a492b35b7f 59a492b36834 59a492b4302c 59a492b43495 59a492b587a6 59a492b5556b 59a492b4a2e1 59a492b42aa6 59a492b3fd6c 59a492b3ee13 59a492b1d565 59a492b1710c 59a492b05b7f 59a492b06909 59a492b06c1c 59a492b0bb7e 59a492b0c5b9 59a492a243e9 59a492a5aa22
```

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30510#discussion_r1747656292

> Did you move these out of here because you'd otherwise get the following error:

Yes, that's the reason. It doesn't seem to be possible to access `src/ipc/` files from the files in the `src/test/` directory by writing `../ipc/` when the `capnp compile` source prefix is `src/test/`, but it does work if the source prefix is `src/`.

I think a different way to work around this would be to pass `--import-path=/path/to/src` to `capnp compile` and then change `using Mining = import "../ipc/capnp/mining.capnp"` to `using Mining = import "/ipc/capnp/mining.capnp", but this would probably be more confusing.

Reply:
That does not really seems less clear to me, but maybe I'm missing something. Using the `--import-path` argument seems akin to the `-I` option I'd pass to a compiler.

Reply:
I just think if you see:

```capnp
using Mining = import "../ipc/capnp/mining.capnp";
```

it is very obvious what file is being imported while if you see

```capnp
using Mining = import "/ipc/capnp/mining.capnp"
```

it's less clear what file is being imported because you don't know what the root path is. This is also inconsistent with standard uses of the root path which have package prefixes "capnp" for "mp":

```capnp
using Cxx = import "/capnp/c++.capnp";
using Proxy = import "/mp/proxy.capnp"
```

We would also be polluting the root namespace with our own files (literally every file in src/) so could cause conflict with other files that are being imported.

If we could include a package name like:

```c++
using Mining = import "/bitcoin/ipc/capnp/mining.capnp"
```

This approach would seem more reasonable to me, but I don't think there is a way to do that with our current directory structure. I think just using relative paths is fine here and probably the way to go.